add-on/chrome/bootstrap.js

```diff
@@ -709,26 +725,55 @@ var WindowListener = {
       },
 
       /**
+       * Set correct strings for infobar notification based on if paused or empty.
+       */
+
+      _setInfoBarStrings: function(nonOwnerParticipants, sharePaused) {
+        let infoStrings = {};
+        if (nonOwnerParticipants) {
+          let stopSharingMessage = this._getString("infobar_screenshare_stop_sharing_message2");
+          let sharingMessage = this._getString("infobar_screenshare_browser_message3");
+          // more than one participant
+          infoStrings.message = sharePaused ? stopSharingMessage : sharingMessage;
+        } else {
+          // empty room
+          let stopSharingNoGuestMessage = this._getString("infobar_screenshare_stop_no_guest_message");
+          let sharingNoGuestMessage = this._getString("infobar_screenshare_no_guest_message");
+          infoStrings.message = sharePaused ? stopSharingNoGuestMessage : sharingNoGuestMessage;
+        }
+        if (sharePaused) {
+          infoStrings.label = this._getString("infobar_button_restart_label2");
+          infoStrings.accesskey = this._getString("infobar_button_restart_accesskey");
+        } else {
+          infoStrings.label = this._getString("infobar_button_stop_label2");
+          infoStrings.accesskey = this._getString("infobar_button_stop_accesskey");
+        }
+        return infoStrings;
+      },
+
+      /**
+       * Indicates if tab sharing is paused.
+       * Set by tab pause button, startBrowserSharing and stopBrowserSharing.
+       * When link generator(owner) enters room we are sharing tabs by default.
+       * Set to default here as startBrowserSharing is called too late for
+       * infobar notification state.
+       */
+      _browserSharePaused: false,
+
+      /**
        * Shows an infobar notification at the top of the browser window that warns
        * the user that their browser tabs are being broadcasted through the current
        * conversation.
+       * @param  {Boolean} optional nonOwnerParticipants is true if a guest is present in the room
```

Comment:
And please audit all callers that aren't passing in a parameter or executing with null set so be sure that something reasonable happens.  If it's possible to eliminate those callers, even better.